TFNetworkLayer.py

```diff
@@ -1928,12 +1928,15 @@ def get_out_data_from_opts(
     out_type = input_data.get_kwargs()
     out_type["name"] = "%s_output" % name
     axis_wo_batch = input_data.get_batch_axis_excluding_batch(axis)
+    feature_dim_wo_batch = input_data.feature_dim_axis_excluding_batch
+    if feature_dim_wo_batch is None:
+      feature_dim_wo_batch = input_data.get_batch_axis_excluding_batch(len(out_type["shape"]) - 1)
```

Comment:
This is wrong. If it is `feature_dim_axis is None`, then for this `Data` instance, it is esp marked that it does not have a feature axis. Maybe you confuse this to `NotSpecified` (but this is only for `feature_dim_axis_or_not_specified`).
You cannot just assume that the last axis might be a feature axis. (Btw, `out_type["shape"]` is without batch, so using `get_batch_axis_excluding_batch` is also wrong here.)